packages/components/src/lib/controllers/app-controller.js

```diff
@@ -1573,71 +1364,53 @@ If no URLS are provided, will reset grammar data.
     }
   }
 
-  /**
-   * Handles a change of a single option from UI options.
-   *
-   * @param {string} name - A name of an option.
-   * @param {string | value} value - A new value of an options.
-   */
-  uiOptionChange (name, value) {
-    let uiOptions = this.api.settings.getUiOptions() // eslint-disable-line prefer-const
-    // TODO this should really be handled within OptionsItem
-    // the difference between value and textValues is a little confusing
-    // see issue #73
-    const nonTextFeatures = [
-      'fontSize',
-      'hideLoginPrompt',
-      'maxPopupWidth',
-      'mouseMoveDelay',
-      'mouseMoveAccuracy',
-      'enableMouseMoveLimitedByIdCheck',
-      'mouseMoveLimitedById',
-      'forceMouseMoveGoogleDocs'
-    ]
-    if (nonTextFeatures.includes(name)) {
-      uiOptions.items[name].setValue(value)
-    } else {
-      uiOptions.items[name].setTextValue(value)
+  applyResourceOption (name, value) {
+    // grouped setting are referenced under Options object
+    // by the parsed name but each individual setting in a group is referenced
+    // by its fullname (with version and groupname appended)
+    // multivalued settings are handled in the Options.setTextValue method which can take
+    // an array or an individual text value
+    const baseKey = Options.parseKey(name)
+    if (baseKey.name === 'grammars') {
+      this.initGrammar(baseKey.group)
     }
-    this.applyUIOptions(name)
   }
 
   /**
    * Updates the state of the UI to match the options settings.
    *
    * @param {string} settingName - The name of the setting to apply.
+   * @param value
    */
-  applyUIOptions (settingName) {
-    const uiOptions = this.api.settings.getUiOptions()
-
+  applyUIOption (settingName, value) {
     switch (settingName) {
       case 'fontSize':
-        this.api.ui.setFontSize()
+        this.api.ui.applyFontSize(value)
         break
       case 'panelPosition':
-        this.store.commit('panel/setPosition', uiOptions.items.panelPosition.currentValue)
+        this.store.commit('panel/setPosition', value)
         break
       case 'verboseMode':
-        this.logger.setVerboseMode(uiOptions.items.verboseMode.currentValue === 'verbose')
+        this.logger.setVerboseMode(this.api.settings.isInVerboseMode())
```

Comment:
this is weird. Why don't we just used the passed value?

Reply:
We had a function called `verboseMode()` that was returning true or false. I renamed it to `isInVerboseMode()` because I thought it would be clearer. Then I decided to use this function (since we use it in many other places) to provide true or false value to the `setVerboseMode()` method. Does it make sense? Or would it be better to leave it as it was before?

Reply:
I think it's just a little strange that we wouldn't just test the value now that we're passing the value in here. Previously this code didn't have access to the values and had to retrieve them from the settings object. But now that the value is directly accessible it's hard to understand why we have to have the extra method call.

Reply:
@balmas, can you please clarify what value are you talking about? Is it what formerly was `uiOptions.items.verboseMode.currentValue`? It is now part of the `_uiOptions` within the Settings Controller. It is private by convention (starts with `_`) and it is not supposed to be accessed from the outside of the Settings Controller (i.e. the App Controller should not access it directly). `isInVerboseMode()`, on the other hand, is part of the Settings' public API, and should be used by other objects to access the value. Or is it some other value you're talking about?
Sorry, I'm a little confused on this.

Reply:
The method is this:

```
  applyUIOption (settingName, value) {
    switch (settingName) {
      case 'fontSize':
        this.api.ui.applyFontSize(value)
        break
      case 'panelPosition':
        this.store.commit('panel/setPosition', value)
        break
      case 'verboseMode':
        this.logger.setVerboseMode(this.api.settings.isInVerboseMode())
        break
      case 'hideLoginPrompt':
        if (this.api.auth) {
          this.store.commit('auth/setHideLoginPrompt', value)
        }
        break
      case 'mouseMoveDelay':
        if (this.selc && this.isMousemoveEnabled) {
          this.selc.updateParamsForAll({ mouseMoveDelay: value })
        }
        break
      case 'mouseMoveAccuracy':
        if (this.selc && this.isMousemoveEnabled) {
          this.selc.updateParamsForAll({ mouseMoveAccuracy: value })
        }
        break
      case 'enableMouseMoveLimitedByIdCheck':
        if (this.selc && this.isMousemoveEnabled) {
          this.selc.updateParamsForAll({ enableMouseMoveLimitedByIdCheck: value })
        }
        break
      case 'forceMouseMoveGoogleDocs':
        if (this.selc) {
          // An event must be replaced because a change of this setting may cause an event change
          this.selc.replaceEventForAll(...this.textSelectorParams)
        }
        break
    }
  }
```

Is it not true that `this.api.settings.isInVerboseMode()` will return the same value as the `value` method parameter?

Reply:
Thanks for explaining, understood it now!
It's not so straightforward, unfortunately, because the value that we receive there is either `verbose` or `normal`, not `true` or `false`. The App Controller is not supposed to have any knowledge of what those values might be, and thus it cannot convert them to the boolean. Does such an approach makes sense? What do you think?

Reply:
ahh, yes ok. This makes sense now. let's leave it as you have it. Thank you. (Maybe we should just add a comment though, so that the next time I ask you we can find the answer easily :-)

Reply:
Good idea, will add a comment about that!